navigation/menu_cache.go

```diff
@@ -48,17 +48,7 @@ type menuCache struct {
 
 // menuEqual checks if two menus are equal.
 func menuEqual(m1, m2 Menu) bool {
-	if len(m1) != len(m2) {
-		return false
-	}
-
-	for i := range m1 {
-		if m1[i] != m2[i] {
-			return false
-		}
-	}
-
-	return true
+	return slices.Equal(m1, m2)
 }
```

Comment:
Thanks for this. This would be even better if you got rid of the menuEqual and compareStringSlices funcs (they're only used in a few places.